test_generate_key fails

Latest Homebrew version of Riak used.

```
$ SKIP_LUWAK=1 SKIP_SEARCH=1 python riak/tests/test_all.py 
ss......ssss....F..........................s....ssss.....ss....ssssssssssssssssssssssssssssssssssssssssssssss.......
======================================================================
FAIL: test_generate_key (__main__.RiakHttpTransportTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "riak/tests/test_all.py", line 935, in test_generate_key
    self.assertEqual(len(bucket.get_keys()), 1)
AssertionError: 2 != 1

----------------------------------------------------------------------
Ran 116 tests in 1.417s

FAILED (failures=1, skipped=59)
```

Test method:

```
def test_generate_key(self):
    # Ensure that Riak generates a random key when
    # the key passed to bucket.new() is None.
    bucket = self.client.bucket('random_key_bucket')
    for key in bucket.get_keys():
        bucket.get(str(key)).delete()
    o = bucket.new(None, data={})
    self.assertIsNone(o.get_key())
    o.store()
    self.assertIsNotNone(o.get_key())
    self.assertNotIn('/', o.get_key())
    self.assertEqual(len(bucket.get_keys()), 1)
```

--- a/riak/tests/test_all.py
+++ b/riak/tests/test_all.py
@@ -126,6 +126,19 @@ class BaseTestCase(object):
         self.assertRaises(TypeError, bucket.new, 'foo', u'éå')
         self.assertRaises(TypeError, bucket.new, 'foo', u'éå')
 
+    def test_generate_key(self):
+        # Ensure that Riak generates a random key when
+        # the key passed to bucket.new() is None.
+        bucket = self.client.bucket('random_key_bucket')
+        existing_keys = bucket.get_keys()
+        o = bucket.new(None, data={})
+        self.assertIsNone(o.get_key())
+        o.store()
+        self.assertIsNotNone(o.get_key())
+        self.assertNotIn('/', o.get_key())
+        self.assertNotIn(o.get_key(), existing_keys)
+        self.assertEqual(len(bucket.get_keys()), len(existing_keys) + 1)
+
     def test_binary_store_and_get(self):
         bucket = self.client.bucket('bucket')
         # Store as binary, retrieve as binary, then compare...
@@ -1017,19 +1030,6 @@ class RiakHttpTransportTestCase(BaseTestCase, MapReduceAliasTestMixIn, unittest.
         bucket = self.client.bucket("bucket")
         o = bucket.new("foo", "bar").store(return_body=False)
         self.assertEqual(o.vclock(), None)
-
-    def test_generate_key(self):
-        # Ensure that Riak generates a random key when
-        # the key passed to bucket.new() is None.
-        bucket = self.client.bucket('random_key_bucket')
-        for key in bucket.get_keys():
-            bucket.get(str(key)).delete()
-        o = bucket.new(None, data={})
-        self.assertIsNone(o.get_key())
-        o.store()
-        self.assertIsNotNone(o.get_key())
-        self.assertNotIn('/', o.get_key())
-        self.assertEqual(len(bucket.get_keys()), 1)
 
     def test_too_many_link_headers_shouldnt_break_http(self):
         bucket = self.client.bucket("bucket")

